backend/app/controllers/secondary_donations_controller.rb

```diff
@@ -2,14 +2,15 @@
 
 class SecondaryDonationsController < ApplicationController
   before_action :authenticate_admin!, except: %i[create]
-  wrap_parameters format: :json, include: %w[amount couponId campaignCharityId]
 
   def index
     @secondary_donations = SecondaryDonation.all
   end
 
   def create
-    @secondary_donation = SecondaryDonation.create!(secondary_donation_params)
+    @secondary_donation = SecondaryDonation.new(secondary_donation_params)
+    @secondary_donation.campaign_charity = CampaignCharity.find(params[:campaign_charity_id])
+    @secondary_donation.save!
```

Comment:
I see that you are manually linking stuff.

Question: Why is this needed? Doesn't rails know to link it when there is a `campaign_charity_id` present?

Reply:
Same as the above comment - `campaign_charity_id` is not an attribute name thus we have to link the associations manually.